### PR TITLE
Remove redundant line breaks for CJK #587

### PR DESCRIPTION
**Pull request by fangyi-zhou**

### CODE REVIEW

**Comment by fangyi-zhou** [comment]
Any idea how do I deal with the test failures?

File: lib/kramdown/utils/html.rb [hunk 1]
```diff
@@ -68,6 +68,14 @@ def escape_html(str, type = :all)
         str.gsub(ESCAPE_RE_FROM_TYPE[type]) {|m| ESCAPE_MAP[m] || m }
       end
 
+      REDUNDANT_LINE_BREAK_REGEX = /([\p{Han}\p{Hiragana}\p{Katakana}]+)\n([\p{Han}\p{Hiragana}\p{Katakana}]+)/u
+      def fix_cjk_line_break(str)
+        while str.match(REDUNDANT_LINE_BREAK_REGEX)
+          str = str.gsub(REDUNDANT_LINE_BREAK_REGEX, '\1\2')
```
Comment (gettalong) [inline lib/kramdown/utils/html.rb]: Does a simple `str.gsub!(REDUNDANT_LINE_BREAK_REGEX, '\1\2')` not work?

**Comment by gettalong** [comment]
You need to adapt `test/test_files.rb` to suppress the tests that cannot be green, like `TestFiles#test_test/testcases/cjk-line-break_text_to_latex_compilation`.

**Comment by gettalong** [comment]
Thank you - I rebased and merged your changes.

**Pull request closed without merging**